gateway-server/src/main/java/org/apache/knox/gateway/services/token/impl/AliasBasedTokenStateService.java

```diff
@@ -125,11 +126,27 @@ protected boolean isUnknown(final String token) {
     return isUnknown;
   }
 
+  @Override
+  protected void removeRevokedExpiredToken(final String token) {
+    if (isUnknown(token)) {
+      log.unknownToken(getTokenDisplayText(token));
+      throw new IllegalArgumentException("Unknown or revoked token.");
+    }
+
+    try {
+      aliasService.removeAliasForCluster(AliasService.NO_CLUSTER_NAME, token);
+      aliasService.removeAliasForCluster(AliasService.NO_CLUSTER_NAME,token + "--max");
+    } catch (AliasServiceException e) {
+      log.failedToUpdateTokenExpiration(e);
+    }
+
+  }
+
   @Override
   protected void updateExpiration(final String token, long expiration) {
     if (isUnknown(token)) {
       log.unknownToken(getTokenDisplayText(token));
-      throw new IllegalArgumentException("Unknown token.");
+      throw new IllegalArgumentException("Unknown or revoked token.");
```

Comment:
I think we can continue treating them simply as "unknown", rather than giving the illusion of distinction.